### PR TITLE
Fastelect

### PR DESCRIPTION
**Pull request by markhannum**
This code reduces election time for a 5-node cluster to less than two seconds.  There were a few bugs, but the main change is to set a random election timeout from each node.

### CODE REVIEW

**Comment by riverszhang89** [comment]
@roborivers Please /runtests for this pull request.

File: berkdb/rep/rep_method.c [hunk 1]
```diff
@@ -346,11 +347,11 @@ __rep_start(dbenv, dbt, gen, flags)
                 __rep_set_gen(dbenv, __func__, __LINE__, rep->recover_gen + 1);
             }
 			if (F_ISSET(rep, REP_F_MASTERELECT)) {
-				__rep_elect_done(dbenv, rep, 0);
+				__rep_elect_done(dbenv, rep, 0, __func__, __LINE__);
 				F_CLR(rep, REP_F_MASTERELECT);
 			}
 			if (rep->egen <= rep->gen)
-				rep->egen = rep->gen + 1;
+                __rep_set_egen(dbenv, __func__, __LINE__, rep->gen + 1);
```
Comment (markhannum) [inline berkdb/rep/rep_method.c]: fix tabs

File: berkdb/rep/rep_method.c [hunk 2]
```diff
@@ -892,6 +893,92 @@ __rep_set_rep_transport(dbenv, eid, f_send)
...
+		goto err;
+
+	curlsn = lastlsn;
+
+	LOGCOPY_32(&rectype, rec.data);
+
+	while (ret == 0 && rectype != DB___txn_regop_gen && rectype !=
+			DB___txn_regop_rowlocks) {
+		if ((ret = __log_c_get(logc, &curlsn, &rec, DB_PREV)) == 0)
+			LOGCOPY_32(&rectype, rec.data);
```
Comment (markhannum) [inline berkdb/rep/rep_method.c]: Punt on txn_regop

**Comment by akshatsikarwar** [comment]
fast

**Pull request merged**